emcache/connection_pool.py

```diff
@@ -261,6 +272,9 @@ async def _create_new_connection(self, backoff=None, retries=None) -> None:
             error = True
         except asyncio.CancelledError:
             logger.info(f"{self} create connection stopped, connection pool is closing")
+        except (AuthenticationError, AuthenticationNotSupportedError) as exc:
+            logger.warning(f"{self} new connection could not be created, failed authentication!")
+            self._waiters[-1].set_exception(exc)
```

Comment:
Better if we just piggy back on this [function](https://github.com/emcache/emcache/pull/146/files#diff-ab7bf183d9513e4b587ce6c652d5ea3dca31e54224d0970690fc7f4ea1060ac5R259), we can extent de signature with a new `connection_error=None` by default, but when we have an errors like this one we call the function 

```
self._wakeup_next_waiter_or_append_to_unused(None, connection_error=exc)
```

Later on we can modify the function for making the `set_exception` during the waiter iteration if and only if the `connection_error` is not None.